runtime/java/src/common/OriginContext.java

```diff
@@ -0,0 +1,136 @@
+package common;
+
+import core.*;
+import java.util.*;
+import common.exceptions.*;
+
+
+/**
+ * Implementation of "the stuff on the left of the turnstile" that needs to be
+ *  passed along for origin tracking. This would be a value type if Java
```

Comment:
Might be good to open a bug about this - things we would change if/when Java gets value types (https://openjdk.java.net/jeps/169.)  Not urgent.